subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/RepositoryHandler.java

```diff
@@ -88,7 +88,17 @@
      * @since 4.4
      */
     @Incubating
-    ArtifactRepository gradlePluginPortal();
+    MavenArtifactRepository gradlePluginPortal();
```

Comment:
I do not think there is a need to make this type more specific and sadly it is a binary breaking change. So please revert it back to the `ArtifactRepository`